runtime/drivers/duckdb/olap.go

```diff
@@ -181,6 +186,344 @@ func (c *connection) EstimateSize() (int64, bool) {
+	}
+	if view {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("ALTER VIEW %s RENAME TO %s", safeSQLName(oldName), safeSQLName(newName)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+	if !c.config.TableAsView {
+		return c.Exec(ctx, &drivers.Statement{
+			Query:       fmt.Sprintf("ALTER TABLE %s RENAME TO %s", safeSQLName(oldName), safeSQLName(newName)),
+			Priority:    100,
+			LongRunning: true,
+		})
+	}
+
+	// ignore query cancellations
+	ctx = context.WithValue(context.Background(), connCtxKey{}, connFromContext(ctx))
+	newSrcDir := filepath.Join(filepath.Dir(c.config.DBFilePath), newName)
+	// new name should not exist
+	exist := dirExist(newSrcDir)
+	if exist {
+		return fmt.Errorf("table %q already exist", newName)
+	}
+
+	oldVersion, exist, err := c.tableVersion(oldName)
+	if err != nil {
+		return err
+	}
+	if !exist {
+		return fmt.Errorf("table %q does not exist", oldName)
+	}
+
+	// drop view
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DROP VIEW IF EXISTS %s", safeSQLName(oldName)),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// detach db
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("DETACH %s", safeSQLName(dbName(oldName, oldVersion))),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// rename dir
+	oldSrcDir := filepath.Join(filepath.Dir(c.config.DBFilePath), oldName)
+	err = os.Rename(oldSrcDir, newSrcDir)
+	if err != nil {
+		return err
+	}
+
+	// rename filename
+	newVersion := fmt.Sprint(time.Now().UnixMilli())
+	newDB := dbName(newName, newVersion)
+	newFilePath := filepath.Join(newSrcDir, fmt.Sprintf("%s.db", newVersion))
+	err = os.Rename(filepath.Join(newSrcDir, fmt.Sprintf("%s.db", oldVersion)), newFilePath)
+	if err != nil {
+		return err
+	}
+
+	// attach new db
+	err = c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("ATTACH '%s' AS %s", newFilePath, safeSQLName(newDB)),
+		Priority:    100,
+		LongRunning: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	// change view query
+	return c.Exec(ctx, &drivers.Statement{
+		Query:       fmt.Sprintf("CREATE OR REPLACE VIEW %s AS SELECT * FROM %s.default", safeSQLName(newName), safeSQLName(newDB)),
+		Priority:    100,
+		LongRunning: true,
+	})
+}
+
+func (c *connection) tableVersion(name string) (string, bool, error) {
+	sourceDir := filepath.Join(filepath.Dir(c.config.DBFilePath), name)
+	files, err := os.ReadDir(sourceDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", false, nil
+		}
+		return "", false, err
+	}
+	if len(files) == 0 {
+		return "", false, nil
+	}
+	// os.ReadDir returns files in sorted by name asc
+	// latest version is the last file
+	for i := len(files) - 1; i >= 0; i-- {
+		file := files[i]
+		name, _, found := strings.Cut(file.Name(), ".db")
+		if !found { // .wal or .tmp file
+			continue
+		}
+		return name, true, nil
+	}
+	return "", false, nil
+}
```

Comment:
I think using `version.txt` would be much simpler and avoid various consistency issues around renames (like if they fail halfway through moving files)